Lesson17

Tatiana-Mandrukevich opened the pull request.

Tatiana-Mandrukevich committed:
Lesson17

qa29onl commented on src/test/java/tests/AddRemoveElements.java:
```diff
@@ -0,0 +1,38 @@
+    //1. Open browser, site http://the-internet.herokuapp.com/add_remove_elements/
+    //2. Click "Add Element" button 2 times
+    //3. Click "Delete" button
+    //4. Check number of elements Delete
```
клаааасс, забыла сказать, чтобы все так делали, молодец)

qa29onl commented on src/test/java/tests/AddRemoveElements.java:
```diff
@@ -0,0 +1,38 @@
+        List<WebElement> listButtonDelete = driver.findElements(By.xpath("//*[@onclick=\"deleteElement()\"]"));
+        int numberOfElementsDelete = listButtonDelete.size();
+
+        Integer expectedResult = 1;
```
можно без этого, сразу в assert закинуть 1

qa29onl commented on src/test/java/tests/AddRemoveElements.java:
```diff
@@ -0,0 +1,38 @@
+        int numberOfElementsDelete = listButtonDelete.size();
+
+        Integer expectedResult = 1;
+        Integer actualResult = numberOfElementsDelete;
```
нет смысла создавать еще одну переменную, можно в ассерт сразу закинуть numberOfElementsDelete

qa29onl commented on src/test/java/tests/Checkboxes.java:
```diff
@@ -0,0 +1,59 @@
+    //6. Uncheck the second checkbox
+    //7. Check that the second checkbox is unchecked
+
+    public void firstCheckboxTest() {
```
надо поменять название теста например - firstCheckboxChecked
в остальных случаях тоже

qa29onl commented on src/test/java/tests/Dropdown.java:
```diff
@@ -0,0 +1,48 @@
+    //5. Select the second element
+    //6. Check that the second element is selected
+
+    public void dropdownTest() {
```
через класс Select лучше работать с дропдаунами

qa29onl commented on src/test/java/tests/Dropdown.java:
```diff
@@ -0,0 +1,48 @@
+
+        boolean expectedResult = false;
+        boolean actualResult = listDropdownElements.isEmpty();
+        Assert.assertEquals(actualResult, expectedResult);
```
можно убрать строчки 28,29 и запихнуть всё в ассерт сразу, в них не так много информации хранится и нет расчетов сложных

qa29onl commented on src/test/java/tests/Hovers.java:
```diff
@@ -0,0 +1,84 @@
+        String expectedUrl1 = "https://the-internet.herokuapp.com/users/1";
+        String actualUrl1 = driver.getCurrentUrl();
+        Assert.assertEquals(actualUrl1, expectedUrl1);
+//        System.out.println("Текущий URL профиля 1: " + driver.getCurrentUrl());
```
мусор в код не коммитаем

qa29onl commented on src/test/java/tests/Hovers.java:
```diff
@@ -0,0 +1,84 @@
+        Assert.assertEquals(actualUrl1, expectedUrl1);
+//        System.out.println("Текущий URL профиля 1: " + driver.getCurrentUrl());
+
+        driver.navigate().back();
```
класс)

qa29onl commented on src/test/java/tests/Hovers.java:
```diff
@@ -0,0 +1,84 @@
+    //5. Check that there is no 404 error (в чате решили, что можно заменить на Assert.assertEquals(driver.getCurrentUrl(), expectedUrl);)
+    //6. Repeat for each profile
+
+    public void hoversTest() {
```
на hover делаем 3 разных теста, для каждого профиля отдельно

qa29onl commented on src/test/java/tests/NotificationMessages.java:
```diff
@@ -0,0 +1,57 @@
+            }
+        } catch (Exception e) {
+            System.out.println("Произошла ошибка: " + e.getMessage());
+        }
```
очень сложная логика, нужно сделать проще, без циклов

qa29onl commented on src/test/java/tests/SortableDataTables.java:
```diff
@@ -0,0 +1,34 @@
+        List<String> expectedResult = List.of(new String[] {"Smith", "John", "jsmith@gmail.com", "$50.00", "http://www.jsmith.com"});
+
+        List<String> actualResult = new LinkedList<>();
+        for (int i = 1; i < 6; i++) {
```
expectedResult.size() вместо 6 лучше использовать, вдруг размер массива поменяется
а так отлично)

qa29onl commented on src/test/java/tests/Typos.java:
```diff
@@ -0,0 +1,29 @@
+        Assert.assertEquals(actualResult, expectedResult);
+
+        String expectedResult2 = "Sometimes you'll see a typo, other times you won't.";
+        String actualResult2 = driver.findElement(By.xpath("//*/p[2]")).getText();
```
//p[2]

Tatiana-Mandrukevich committed:
Fixed the following tests:
1. AddRemoveElements
2. Checkboxes
3. Dropdown
4. Hovers
5. NotificationMessages
6. SortableDataTables
7. Typos
